tests/auth_rust/tests/test_auth_demo.rs

```diff
@@ -203,9 +357,237 @@ fn rsa_verify_failed() {
+            assert_eq!(msg_hex.len(), 64);
+
+            let mut temp2: BytesMut = BytesMut::with_capacity(message_magic.len() + msg_hex.len());
+            temp2.put(Bytes::from(message_magic.to_vec()));
+            temp2.put(Bytes::from(hex::encode(message)));
+
+            let mut md = Md::new(Type::Sha256).unwrap();
+            md.update(temp2.freeze().to_vec().as_slice())
+                .expect("md btc failed");
+            let mut msg: [u8; 32] = [0; 32];
+            md.finish(&mut msg).expect("md btc sha256 finish");
+
+            let mut md = Md::new(Type::Sha256).unwrap();
+            md.update(&msg).expect("md btc new message failed");
+            md.finish(&mut msg)
+                .expect("md btc convert message finish failed");
+
+            H256::from(msg)
+        }
+        fn sign(&self, msg: &H256) -> Bytes {
+            BitcoinAuth::btc_sign(msg, &self.0.privkey, self.0.compress)
+        }
+    }
+
+    let privkey = Generator::random_privkey();
+    let auth = Box::new(BtcConverFaileAuth {
+        0: BitcoinAuth {
+            privkey,
+            compress: true,
+        },
+    });
+
+    let verify_result = unit_test_with_auth(auth, false);
+    assert_script_error(verify_result.unwrap_err(), AuthErrorCodeType::Mismatched);
+}
+
+#[test]
+fn convert_doge_error() {
+    struct DogeConverFaileAuth(DogecoinAuth);
+    impl Auth for DogeConverFaileAuth {
+        fn get_pub_key_hash(&self) -> Vec<u8> {
+            BitcoinAuth::get_btc_pub_key_hash(&self.0.privkey, self.0.compress)
+        }
+        fn get_algorithm_type(&self) -> u8 {
+            AlgorithmType::Bitcoin as u8
+        }
+        fn convert_message(&self, message: &[u8; 32]) -> H256 {
+            let message_magic = b"\x18Bitcoin Signed Xessage:\n\x40";
+            let msg_hex = hex::encode(message);
+            assert_eq!(msg_hex.len(), 64);
+
+            let mut temp2: BytesMut = BytesMut::with_capacity(message_magic.len() + msg_hex.len());
+            temp2.put(Bytes::from(message_magic.to_vec()));
+            temp2.put(Bytes::from(hex::encode(message)));
+
+            let mut md = Md::new(Type::Sha256).unwrap();
+            md.update(temp2.freeze().to_vec().as_slice())
+                .expect("md btc failed");
+            let mut msg: [u8; 32] = [0; 32];
+            md.finish(&mut msg).expect("md btc sha256 finish");
+
+            let mut md = Md::new(Type::Sha256).unwrap();
+            md.update(&msg).expect("md btc new message failed");
+            md.finish(&mut msg)
+                .expect("md btc convert message finish failed");
+
+            H256::from(msg)
+        }
+        fn sign(&self, msg: &H256) -> Bytes {
+            BitcoinAuth::btc_sign(msg, &self.0.privkey, self.0.compress)
+        }
+    }
+
+    let privkey = Generator::random_privkey();
+    let auth = Box::new(DogeConverFaileAuth {
+        0: DogecoinAuth {
+            privkey,
+            compress: true,
+        },
```

Comment:
add test case when `recid` is invalid:  
https://github.com/nervosnetwork/ckb-production-scripts/blob/ec2a7f1ccac27a7b73f5c95be57a48021cc9d4e3/c/auth.c#L122